lib/client-assets.php

```diff
@@ -715,8 +715,19 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	array_push( $word_count_script->deps, 'wp-utils' );
 
 	global $post;
+	// On admin pages, global $post is the only post.
+	if ( is_admin() ) {
+		$post_to_edit = $post;
+	} else {
+		// On front-end editing, global $post might be last post in the_loop
```

Comment:
Should the `$post_to_edit` logic happen here if we want to be able to reuse `gutenberg_editor_scripts_and_styles` across different rendering contexts? Wondering if post (or post ID) should be an argument to the function itself, called from separate hook handlers.